src/lifecycle/properties.js

```diff
@@ -90,50 +58,60 @@ function property (elem, name, prop) {
     // value internally because the default getter may still be used to return
     // that value. Even if it's not, we use it to reference the old value which
     // is useful information for the setter.
-    if (internalSetter) {
-      internalSetter.call(this, newValue, oldValue);
+    if (prop.update) {
+      prop.update.call(this, newValue, oldValue);
+    }
+
+    // If we are emitting notify the element of the change.
+    if (prop.emit) {
+      emit(this, prop.emit, {
```

Comment:
This is the addition of the `skate.property` event. @marcoms will probably be happy about this. I think it's not disputed that we do need this, but I would like to discuss whether or not we should be emitting:

1. `skate.property`
2. `skate.property.[name]`
3. one or both of the above

A case for 1 is a template that wants to listen to all property updates.

A case for 2 is possibly performance so handlers aren't invoked for every property change. The downside is that you *must* be specific about which properties you want to listen to.

A case for both is flexibility, but performance might suffer when a lot of  state changing is happening.